homeassistant/config.py

```diff
@@ -947,10 +1024,27 @@ async def async_process_component_config(  # noqa: C901
                     p_name,
                     domain,
                 )
+                failed_platforms.add(p_name)
                 continue
 
         platforms.append(p_validated)
 
+    if raise_on_platform_setup_errors and failed_platforms:
+        platforms_list = list(failed_platforms)
+        platform_count = len(platforms_list)
+        platforms_list_effected = ", ".join(platforms_list)
+        raise HomeAssistantError(
+            f"Failed to set up platform{'' if platform_count==1 else ''} "
+            f"{platforms_list_effected} for domain {domain}, check the logs for more information",
+            translation_domain="homeassistant",
+            translation_key="component_config_error",
+            translation_placeholders={
+                "domain": domain,
+                "platforms": platforms_list_effected,
+                "count": str(platform_count),
+            },
+        )
+
```

Comment:
Please remove this from here, this function should instead return `failed_platforms`, then the caller can decide to raise or log.